File: write.java
import com.splunk.*; 
import com.splunk.JobArgs.ExecutionMode;

import java.util.*;
import java.io.*;

public class write {

    public static void main(String[] args) {

		try {
			if(args.length != 2) {
				System.out.println("syntax: write index \"data\"");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);

			Service service = Service.connect(command.opts);

	        Args params = new Args();
	        	//LAB: add name/value pairs for sourcetype, source, and host, setting any static value for each as desired
	        
	        Index index = //LAB: get the index named in args[0]
	        //LAB: submit the data in args[1] to the index, passing the params, and add a date/time stamp to the beginning of the data
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}

File: python/write.py

import splunklib.client as client
import utils
import sys
import datetime

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 3:
    print "Syntax: write.py index \"data\""
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)
 
    index = #LAB: get the index named in sys.argv[1]
    #LAB: submit the data in sys.argv[2] to the index, adding a date/time stamp to the beginning of the data, and set host, source and sourcetype to static values as desired
    
except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))
